docs/src/tutorials/MHMPC.md

```diff
@@ -0,0 +1,229 @@
+    end
+    
+    # Add subgraphs to master graph
+    for subgraph in subgraphs_all
+        add_subgraph(graph_master, subgraph)
+    end
+    
+    # Add linking constraint between subgraphs
+    @linkconstraint(graph_master, [n = 1:(length(subgraphs_all)-1)], nodes_all[n+1][1][:x][1] == nodes_all[n][N_i[n]][:Ax_Bu][1])
+    @linkconstraint(graph_master, [n = 1:(length(subgraphs_all)-1)], nodes_all[n+1][1][:x][2] == nodes_all[n][N_i[n]][:Ax_Bu][2])
+      
+```
+
+### 5. Setting objective and running optimization
+
+The final step is to set the objective and run the optimization. This step is written out in the mathematic formulation and is just written out. The code to run the optimization is straight forward as well. `set_to_node_objectives` is a newer function that is required for solving the optimization problem.
+
+The resulting output is the optimal values for $x, u$. For plotting and running this optimization over a long time period, we return the arrays of state variables and input variables. Recall, the code from step 2 to this step is wrapped in a function.
+
+```julia
+    # Set objective
+    @objective(
+        graph_master, 
+        Min, 
+        sum(node[:u]' * R * node[:u] for node in all_nodes(graph_master)) +
+        sum(node[:x]' * Q * node[:x] for node in all_nodes(graph_master)),
+    )
+    
+    # Run optimization
+    set_to_node_objectives(graph_master)
+    set_optimizer(graph_master, Ipopt.Optimizer)
+    set_optimizer_attribute(graph_master, "print_level", 0) # Silence the optimization output
+    optimize!(graph_master)
+
+    # Get output values
+    vals_x1 = []
+    vals_x2 = []
+    vals_u = []
+
+    for node in all_nodes(graph_master)
+        push!(vals_x1, value(graph_master, node[:x][1]))
+        push!(vals_x2, value(graph_master, node[:x][2]))
+        push!(vals_u, value(graph_master, node[:u][1]))
+    end
+
+    return vals_x1, vals_x2, vals_u
+end
+```
+
+### 6. Run the program
+
+In context of the spring damper system, we want to apply an optimal force $u$ to dampen the system to position $0$, and $0$ velocity. This optimization problem will give values for $u$ at each time discretization node. For our purposes, we choose to run this program an arbitrary $400$ times, once every $0.05$ seconds. Thus every time step ($0.05$ seconds), the future state variables will be parameters to the next optimization problem.
+
+```julia
+function run_sim(T::Int = 401)
+    x_arr = Vector{Vector{Float64}}() # array of state variables at each timepoint
+    u_arr = Vector{Float64}() # array of input variables at each time point
+
+    x = [18.0, -14.0]
+    push!(x_arr, x)
+
+    for _ in 2:T
+        vals_x1, vals_x2, vals_u = MHMPC(x)
+        x = [vals_x1[2], vals_x2[2]]
+        push!(x_arr, x)
+        push!(u_arr, vals_u[1])
+    end
+
+    return x_arr, u_arr
+end
+```
+
+Now, plotting the values of $x$ from our program:
```

Comment:
Maybe at the end of this last code snippet, add the code to call the `run_sim` function. Just something like:
```julia
x_sim, u_sim = run_sim()
```